src/main/java/duke/Task.java

```diff
@@ -0,0 +1,71 @@
+package duke;
+
+import java.time.LocalDate;
+
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+    private LocalDate date;
+
+    /**
+     * Constructs and initializes the attributes of a new Task object.
+     * @param description The description of the task.
+     * @param isDone The status of the task.
+     * @param date The date related to the task.
+     */
+    public Task (String description, boolean isDone, LocalDate date) {
+        this.description = description;
+        this.isDone = isDone;
+        this.date = date;
+    }
+
+    /**
+     * Constructs and initializes the attributes of a new Task object.
+     * @param description The description of the task.
+     * @param isDone The status of the task.
+     */
+    public Task (String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public String getDate() {
+        return date.toString();
+    }
+
+    /**
+     * Make a task to be completed
+     * @return A new task object with the same description and date but is completed
+     */
+    public Task completeTask() {
+        if (this instanceof Deadline) {

```

Comment:
Perhaps instead of using instanceof to discover the object type of the Tasks, maybe u can just define a separate enum class that carries all main types of tasks and then create a variable in Task abstract class for defining the type of the task?

Reply:
Good idea! Thanks for the advice!